File: go/examples/basic_listener.go
package examples

import (
	"fmt"
	"log"
	"net"
)

func Listener() {
	port := ":3090"
	listener, err := net.Listen("tcp", port)
	if err != nil {
		panic(err)
	}

	fmt.Println("Listening on port ", port)

	for {
		conn, err := listener.Accept()
		if err != nil {
			log.Fatalf("Error while accepting socket connection %v", err)
		}
		fmt.Println(conn)
	}
}

File: java/src/main/java/org/example/BasicListener.java
package org.example;

import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

public class BasicListener {

    public static void main(String[] args) throws IOException {
        int port = 3090;
        try (final ServerSocket serverSocket = new ServerSocket(port)) {
            System.out.println("Listening on port 3090 ");
            while (true) {
                Socket connection = serverSocket.accept();
                System.out.println(connection);
            }
        }
    }
}
